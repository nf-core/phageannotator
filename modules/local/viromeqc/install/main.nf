process VIROMEQC_INSTALL {
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-b28a1a551d380ce8d57f9d83894ccb9559b44404:08a4cf815fcef0080ede5b3633202cbda8edf59b-0':
        'biocontainers/mulled-v2-b28a1a551d380ce8d57f9d83894ccb9559b44404:08a4cf815fcef0080ede5b3633202cbda8edf59b-0' }"

    output:
    path("${prefix}/")  , emit: viromeqc_index
    path "versions.yml" , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "viromeqc_index"
    """
    viromeQC.py \\
        --install \\
        --index_dir ${prefix} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        viromeqc: \$(echo \$(viromeQC.py --version 2>&1 | sed 's/^.*viromeQC.py../ //; s/Using.*\$//' ))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "viromeqc_index"
    """
    mkdir -p ${prefix}
    touch ${prefix}/SILVA_132_LSURef_tax_silva.clean.1.bt2
    touch ${prefix}/SILVA_132_LSURef_tax_silva.clean.2.bt2
    touch ${prefix}/SILVA_132_LSURef_tax_silva.clean.3.bt2
    touch ${prefix}/SILVA_132_LSURef_tax_silva.clean.4.bt2
    touch ${prefix}/SILVA_132_LSURef_tax_silva.clean.rev.1.bt2
    touch ${prefix}/SILVA_132_LSURef_tax_silva.clean.rev.2.bt2
    touch ${prefix}/SILVA_132_SSURef_Nr99_tax_silva.clean.1.bt2
    touch ${prefix}/SILVA_132_SSURef_Nr99_tax_silva.clean.2.bt2
    touch ${prefix}/SILVA_132_SSURef_Nr99_tax_silva.clean.3.bt2
    touch ${prefix}/SILVA_132_SSURef_Nr99_tax_silva.clean.4.bt2
    touch ${prefix}/SILVA_132_SSURef_Nr99_tax_silva.clean.rev.1.bt2
    touch ${prefix}/SILVA_132_SSURef_Nr99_tax_silva.clean.rev.2.bt2
    touch ${prefix}/amphora_bacteria.dmnd
    touch ${prefix}/amphora_bacteria_294.dmnd

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        viromeqc: \$(echo \$(viromeQC.py --version 2>&1 | sed 's/^.*viromeQC.py //; s/Using.*\$//' ))
    END_VERSIONS
    """
}
