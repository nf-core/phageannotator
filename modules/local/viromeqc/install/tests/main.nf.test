nextflow_process {

    name "Test Process: VIROMEQC_INSTALL"
    script "../main.nf"
    process "VIROMEQC_INSTALL"


    test("No input required") {
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("No input required - stub") {
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
