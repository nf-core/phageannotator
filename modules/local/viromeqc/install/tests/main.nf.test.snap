{
    "No input required": {
        "content": [
            {
                "0": [
                    [
                        "SILVA_132_LSURef_tax_silva.clean.1.bt2:md5,c125f0dd338c049df0c6f3a23b93947b",
                        "SILVA_132_LSURef_tax_silva.clean.2.bt2:md5,4ceda8501a2c8652a7fe18b3367c5268",
                        "SILVA_132_LSURef_tax_silva.clean.3.bt2:md5,8a99411da3111a7ed68ffab5838a8d19",
                        "SILVA_132_LSURef_tax_silva.clean.4.bt2:md5,585f66a0eb9223716d759e1f67c47ff2",
                        "SILVA_132_LSURef_tax_silva.clean.rev.1.bt2:md5,2e34f45d326fe81a99b0764c1792732a",
                        "SILVA_132_LSURef_tax_silva.clean.rev.2.bt2:md5,9a4dfd0f63e6ad874d64aa218225ad7a",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.1.bt2:md5,b9aa098575b0bf2de3907ccb9326f3f3",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.2.bt2:md5,4832a022ffb43a20ced82a7902cc313e",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.3.bt2:md5,a2024419722a96056ca3774b8405567d",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.4.bt2:md5,59911dc532215f8320ece7f64b8e32ed",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.1.bt2:md5,6204db0004c5b2302c7e90c65b5ee902",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.2.bt2:md5,94b967bbeee3a5fb277217fd26c838d1",
                        "amphora_bacteria.dmnd:md5,eda6391a0026045bb70c3f2758775baf",
                        "amphora_bacteria_294.dmnd:md5,fe18f81444543940fca08c8642149b7d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,62af8c4ed30786021d8be4d24c27cc92"
                ],
                "versions": [
                    "versions.yml:md5,62af8c4ed30786021d8be4d24c27cc92"
                ],
                "viromeqc_index": [
                    [
                        "SILVA_132_LSURef_tax_silva.clean.1.bt2:md5,c125f0dd338c049df0c6f3a23b93947b",
                        "SILVA_132_LSURef_tax_silva.clean.2.bt2:md5,4ceda8501a2c8652a7fe18b3367c5268",
                        "SILVA_132_LSURef_tax_silva.clean.3.bt2:md5,8a99411da3111a7ed68ffab5838a8d19",
                        "SILVA_132_LSURef_tax_silva.clean.4.bt2:md5,585f66a0eb9223716d759e1f67c47ff2",
                        "SILVA_132_LSURef_tax_silva.clean.rev.1.bt2:md5,2e34f45d326fe81a99b0764c1792732a",
                        "SILVA_132_LSURef_tax_silva.clean.rev.2.bt2:md5,9a4dfd0f63e6ad874d64aa218225ad7a",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.1.bt2:md5,b9aa098575b0bf2de3907ccb9326f3f3",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.2.bt2:md5,4832a022ffb43a20ced82a7902cc313e",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.3.bt2:md5,a2024419722a96056ca3774b8405567d",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.4.bt2:md5,59911dc532215f8320ece7f64b8e32ed",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.1.bt2:md5,6204db0004c5b2302c7e90c65b5ee902",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.2.bt2:md5,94b967bbeee3a5fb277217fd26c838d1",
                        "amphora_bacteria.dmnd:md5,eda6391a0026045bb70c3f2758775baf",
                        "amphora_bacteria_294.dmnd:md5,fe18f81444543940fca08c8642149b7d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:06:23.877411726"
    },
    "No input required - stub": {
        "content": [
            {
                "0": [
                    [
                        "SILVA_132_LSURef_tax_silva.clean.1.bt2:md5,c125f0dd338c049df0c6f3a23b93947b",
                        "SILVA_132_LSURef_tax_silva.clean.2.bt2:md5,4ceda8501a2c8652a7fe18b3367c5268",
                        "SILVA_132_LSURef_tax_silva.clean.3.bt2:md5,8a99411da3111a7ed68ffab5838a8d19",
                        "SILVA_132_LSURef_tax_silva.clean.4.bt2:md5,585f66a0eb9223716d759e1f67c47ff2",
                        "SILVA_132_LSURef_tax_silva.clean.rev.1.bt2:md5,2e34f45d326fe81a99b0764c1792732a",
                        "SILVA_132_LSURef_tax_silva.clean.rev.2.bt2:md5,9a4dfd0f63e6ad874d64aa218225ad7a",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.1.bt2:md5,b9aa098575b0bf2de3907ccb9326f3f3",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.2.bt2:md5,4832a022ffb43a20ced82a7902cc313e",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.3.bt2:md5,a2024419722a96056ca3774b8405567d",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.4.bt2:md5,59911dc532215f8320ece7f64b8e32ed",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.1.bt2:md5,6204db0004c5b2302c7e90c65b5ee902",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.2.bt2:md5,94b967bbeee3a5fb277217fd26c838d1",
                        "amphora_bacteria.dmnd:md5,eda6391a0026045bb70c3f2758775baf",
                        "amphora_bacteria_294.dmnd:md5,fe18f81444543940fca08c8642149b7d"
                    ]
                ],
                "1": [
                    "versions.yml:md5,62af8c4ed30786021d8be4d24c27cc92"
                ],
                "versions": [
                    "versions.yml:md5,62af8c4ed30786021d8be4d24c27cc92"
                ],
                "viromeqc_index": [
                    [
                        "SILVA_132_LSURef_tax_silva.clean.1.bt2:md5,c125f0dd338c049df0c6f3a23b93947b",
                        "SILVA_132_LSURef_tax_silva.clean.2.bt2:md5,4ceda8501a2c8652a7fe18b3367c5268",
                        "SILVA_132_LSURef_tax_silva.clean.3.bt2:md5,8a99411da3111a7ed68ffab5838a8d19",
                        "SILVA_132_LSURef_tax_silva.clean.4.bt2:md5,585f66a0eb9223716d759e1f67c47ff2",
                        "SILVA_132_LSURef_tax_silva.clean.rev.1.bt2:md5,2e34f45d326fe81a99b0764c1792732a",
                        "SILVA_132_LSURef_tax_silva.clean.rev.2.bt2:md5,9a4dfd0f63e6ad874d64aa218225ad7a",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.1.bt2:md5,b9aa098575b0bf2de3907ccb9326f3f3",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.2.bt2:md5,4832a022ffb43a20ced82a7902cc313e",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.3.bt2:md5,a2024419722a96056ca3774b8405567d",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.4.bt2:md5,59911dc532215f8320ece7f64b8e32ed",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.1.bt2:md5,6204db0004c5b2302c7e90c65b5ee902",
                        "SILVA_132_SSURef_Nr99_tax_silva.clean.rev.2.bt2:md5,94b967bbeee3a5fb277217fd26c838d1",
                        "amphora_bacteria.dmnd:md5,eda6391a0026045bb70c3f2758775baf",
                        "amphora_bacteria_294.dmnd:md5,fe18f81444543940fca08c8642149b7d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:09:07.248281569"
    }
}
